final_prep/mod_zero_hero.rb

```diff
@@ -2,52 +2,121 @@
+
 # Declare two variables - is_human and identity_concealed - assigned to booleans
 
+is_human = true
+identity_concealed = false
 
 # Declare two variables - arch_enemies AND sidekicks
 #   arch_enemies should be an array of at least 3 different enemy strings
 #   sidekicks should be an array of at least 3 different sidekick strings
 
+arch_enemies = ["Dog Boy", "Super Fish", "Alarm Clocks"]
+sidekicks = ["Mouse Toy", "Soft Blanket", "Catnip"]
+
 # Print the first sidekick to your terminal
 
+print sidekicks[0]
+
 # Print the last arch_enemy to the terminal
 
+print arch_enemies[2]
+
 # Write some code to add a new arch_enemy to the arch_enemies array
 
+arch_enemies.push("Vacuum Cleaner")
+
 # Print the arch_enemies array to terminal to ensure you added a new arch_enemey
 
+print arch_enemies
+
 # Remove the first sidekick from the sidekicks array
 
+sidekicks.shift
+
 # Print the sidekicks array to terminal to ensure you added a new sidekick
 
+print sidekicks
+
 # Create a function called assess_situation that takes three arguments - danger_level, save_the_day, bad_excuse
 #   - danger_level should be an integer
-#   - save_the_day should be a string a hero would say once they save the day 
+#   - save_the_day should be a string a hero would say once they save the day
 #   - bad_excuse should be a string a hero would say if they are too afraid of the danger_level
 
+def assess_situation(danger_level, save_the_day, bad_excuse)
+  @danger_level = danger_level
+  @save_the_day = save_the_day
+  @bad_excuse = bad_excuse
+end
+
+def save_the_day
+  puts "Once again Catman has saved the day by napping!"
+end
+
+def bad_excuse
+  puts "This sounds like a job for someone who has already taken a nap!"
+end
+
+def danger_level
+  60
+end
+
+
 # Your function should include an if/else statement that meets the following criteria
 #   - Danger levels that are above 50 are too scary for your hero. Any danger level that is above 50 should result in printing the bad_excuse to the terminal
 #   - Anything danger_level that is between 10 and 50 should result in printing the save_the_day string to the terminal
 #   - If the danger_level is below 10, it means it is not worth your time and should result in printing the string "Meh. Hard pass." to the terminal.
 
+
+if danger_level > 50
+  puts bad_excuse
+elsif danger_level.between?(10, 50)
+  puts save_the_day
+else danger_level < 10
+  puts "Meh. Hard pass."
+end
+
+
 #Test Cases
 announcement = 'Never fear, the Courageous Curly Bracket is here!'
 excuse = 'I think I forgot to lock up my 1992 Toyota Coralla. Be right back.'
 # assess_situation(99, announcement, excuse) > Should print - 'I think I forgot to lock up my 1992 Toyota Coralla. Be right back.'
 #assess_situation(21, announcement, excuse) > should print - 'Never fear, the Courageous Curly Bracket is here!'
 #assess_situation(3, announcement, excuse) > should print - "Meh. Hard pass."
 
+assess_situation(99, announcement, excuse)
+puts excuse
```

Comment:
Here we are invoking (calling) the function `assess_situation` and expecting some sort of text to be printed based on the arguments passed into the function - You don't need to manually print `excuse` or `announcement` or `"Meh, hard pass."` - the function outcome will do that for us.